### PR TITLE
OPSEXP-2849 Create reusable build/test workflow with ACS 7.4 support

### PR DESCRIPTION
**Pull request by pmacius**
### Description

- Created reusable workflow for building and testing
- Added ACS 7.4
- ARM builds only for the current ACS version 23
- fetch-compose.sh script is by default downloading also 7.4-compose file
- Tomcat for versions prior to 23 is build with version 9 (vars set in workflow step)
- Added steps which calculates the files names for helm and compose

### Related Issue

OPSEXP-2849

### Checklist

- [x] My code follows the project's coding standards.
- [ ] I have updated the documentation accordingly.

Documentation will be updated as a part of another task

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -48,277 +41,17 @@ jobs:
...
-        with:
-          name: k8s-logs-${{ matrix.arch }}-${{ matrix.edition }}
-          path: logs
+        version: [23, 74]
+    with:
+      registry: ghcr.io
+      registry_namespace: alfresco
+      tag: ${{ github.event_name == 'pull_request' && format('pr-{0}-v{1}', github.event.pull_request.number, matrix.version) || format('{0}-v{1}', github.ref_name, matrix.version) }}
+      acs_deployment_version: 78132d95e29d7126025e127740d10cf958164947 # 8.6.0-alpha.0 with compose healthcheck
+      acs_version: ${{ matrix.version }}
```
Comment (gionn) [inline .github/workflows/build_and_test.yml:56]: you probably need just tag and acs_version here, hardcode them in the reusable workflow (we don't really need to make them dynamic on caller)

File: .github/workflows/reusuable_build_and_test.yml
Comment (gionn) [inline .github/workflows/reusuable_build_and_test.yml:1]: check the typo in the filename

File: .github/workflows/reusable_build_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,345 @@
+name: CI
```
Comment (gionn) [inline .github/workflows/reusable_build_and_test.yml:1]:
```suggestion
name: Build and Test
run-name: Build v${{ inputs.acs_version }}
```

to make it better:

Screenshot 2024-11-07 at 12 37 44

**Review by gionn — COMMENTED** [review]
everything else LGTM

**Review by gionn — APPROVED** [review]

File: .github/workflows/cleanup_images.yml [hunk 3]
```diff
@@ -53,14 +53,15 @@ jobs:
...
+          PR_TAGS: ${{ format('pr-{0}*', github.event.pull_request.number) }}
         with:
           token: ${{ secrets.DELETE_PACKAGES_GITHUB_TOKEN }}
           owner: ${{ env.ORG }}
           repository: ${{ env.REPO }}
           packages: ${{ env.PACKAGE_NAMES }}
-          delete-tags: ${{ env.PR_TAG }}
+          delete-tags: ${{ env.PR_TAGS }}
           dry-run: false
+          use-regex: true
```
Comment (gionn) [inline .github/workflows/cleanup_images.yml:64]:
```suggestion
```

was remembering wrong, looks like `*` is supported without the regexp switch
https://github.com/dataaxiom/ghcr-cleanup-action?tab=readme-ov-file#delete-tags

**Review by gionn — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -28,6 +28,19 @@ help:
...
+
+ifeq ($(ACS_VERSION), 23)
+  TOMCAT_FIELD := "tomcat10"
+else
+  TOMCAT_FIELD := "tomcat9"
+endif
+
+TOMCAT_MAJOR := $(shell yq e '.${TOMCAT_FIELD}.major' $(TOMCAT_VERSIONS_FILE))
+TOMCAT_VERSION := $(shell yq e '.${TOMCAT_FIELD}.version' $(TOMCAT_VERSIONS_FILE))
+TOMCAT_SHA512 := $(shell yq e '.${TOMCAT_FIELD}.sha512' $(TOMCAT_VERSIONS_FILE))
```
Comment (alxgomz) [inline Makefile:42]: Can you add `yq` in the prereq section of the readme file?

**Review by alxgomz — CHANGES_REQUESTED** [review]

**Review by alxgomz — APPROVED** [review]

**Pull request merged**